plancheck/expect_non_empty_plan_test.go

```diff
@@ -9,26 +9,83 @@ import (
 
 	r "github.com/hashicorp/terraform-plugin-testing/helper/resource"
 	"github.com/hashicorp/terraform-plugin-testing/plancheck"
+	"github.com/hashicorp/terraform-plugin-testing/tfversion"
 )
 
-func Test_ExpectNonEmptyPlan(t *testing.T) {
+func Test_ExpectNonEmptyPlan_OutputChanges_None(t *testing.T) {
 	t.Parallel()
 
-	r.Test(t, r.TestCase{
+	r.UnitTest(t, r.TestCase{
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.SkipBelow(tfversion.Version0_14_0), // outputs before 0.14 always show as created
+		},
+		// Avoid our own validation that requires at least one provider config.
 		ExternalProviders: map[string]r.ExternalProvider{
-			"random": {
-				Source: "registry.terraform.io/hashicorp/random",
+			"terraform": {Source: "terraform.io/builtin/terraform"},
+		},
+		Steps: []r.TestStep{
+			{
+				Config: `output "test" { value = "original" }`,
+			},
+			{
+				Config: `output "test" { value = "new" }`,
+				ConfigPlanChecks: r.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						plancheck.ExpectNonEmptyPlan(),
+					},
+				},
 			},
 		},
+	})
+}
+
+func Test_ExpectNonEmptyPlan_OutputChanges_Error(t *testing.T) {
+	t.Parallel()
+
+	r.UnitTest(t, r.TestCase{
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.SkipBelow(tfversion.Version0_14_0), // outputs before 0.14 always show as created
+		},
+		// Avoid our own validation that requires at least one provider config.
+		ExternalProviders: map[string]r.ExternalProvider{
+			"terraform": {Source: "terraform.io/builtin/terraform"},
+		},
 		Steps: []r.TestStep{
 			{
-				Config: `resource "random_string" "one" {
-					length = 16
+				Config: `output "test" { value = "original" }`,
+			},
+			{
+				Config: `output "test" { value = "original" }`,
+				ConfigPlanChecks: r.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						plancheck.ExpectNonEmptyPlan(),
+					},
+				},
+				ExpectError: regexp.MustCompile(`expected a non-empty plan, but got an empty plan`),
+			},
+		},
+	})
+}
+
+func Test_ExpectNonEmptyPlan_ResourceChanges_None(t *testing.T) {
```

Comment:
Nit: Should `_None` be used here? Seems like it's indicating that there are no resource changes?